### ISSUE TITLE
datachannel.close() segfaults

### ISSUE THREAD

**Issue by silverbucket** [node-webrtc/node-webrtc#292]
Whenever the `datachannel.close()` function is called (which, in turn calls `internalDC.close()`):

Here's the backtrace:

```
Thread 1 "node" received signal SIGSEGV, Segmentation fault.
0x00007ffff0f9610f in node_webrtc::DataChannel::Close(Nan::FunctionCallbackInfo<v8::Value> const&)
    () from /home/nkj/code/node-webrtc/build/wrtc/v0.0.62/Release/node-v48-linux-x64/wrtc.node
(gdb) backtrace
#0  0x00007ffff0f9610f in node_webrtc::DataChannel::Close(Nan::FunctionCallbackInfo<v8::Value> const&) () from /home/nkj/code/node-webrtc/build/wrtc/v0.0.62/Release/node-v48-linux-x64/wrtc.node
#1  0x00007ffff0f959b6 in Nan::imp::FunctionCallbackWrapper(v8::FunctionCallbackInfo<v8::Value> const&) () from /home/nkj/code/node-webrtc/build/wrtc/v0.0.62/Release/node-v48-linux-x64/wrtc.node
#2  0x00000000006fa671 in v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) ()
#3  0x000000000075046b in ?? ()
#4  0x0000000000750b38 in ?? ()
#5  0x0000000000750c9e in ?? ()
#6  0x0000276c6f0092a7 in ?? ()
#7  0x00007fffffffa050 in ?? ()
#8  0x0000276c6f0091e1 in ?? ()
#9  0x00007fffffffa060 in ?? ()
#10 0x0000000300000000 in ?? ()
#11 0x00007fffffffa0c0 in ?? ()
#12 0x0000276c6f35538f in ?? ()
#13 0x00000f1aac3b0ef9 in ?? ()
#14 0x000000e0b9140691 in ?? ()
#15 0x00000f1aac3b0ef9 in ?? ()
#16 0x00001d4fcb5c0651 in ?? ()
#17 0x00001d4fcb5c0651 in ?? ()
#18 0x000000e0b91409f9 in ?? ()
#19 0x00007fffffffa0f0 in ?? ()
#20 0x0000276c6f35515d in ?? ()
#21 0x000000e0b9140991 in ?? ()
#22 0x00001d4fcb5c0651 in ?? ()
#23 0x00003dfb0c497931 in ?? ()
#24 0x000011a7805af0c9 in ?? ()
#25 0x00007fffffffa120 in ?? ()
#26 0x0000276c6f35490d in ?? ()
#27 0x000000e0b9142fe1 in ?? ()
#28 0x00003dfb0c497931 in ?? ()
#29 0x00003dfb0c49cb41 in ?? ()
#30 0x000011a7805af0c9 in ?? ()
#31 0x00007fffffffa150 in ?? ()
#32 0x0000276c6f354590 in ?? ()
#33 0x00001c6f380cda81 in ?? ()
#34 0x00003dfb0c49cb41 in ?? ()
#35 0x00003dfb0c49cbf1 in ?? ()
#36 0x000011a7805af0c9 in ?? ()
#37 0x00007fffffffa190 in ?? ()
#38 0x0000276c6f354263 in ?? ()
#39 0x000011a780504201 in ?? ()
#40 0x00001c6f380cda81 in ?? ()
#41 0x00003dfb0c49cbf1 in ?? ()
#42 0x00001c6f380cda81 in ?? ()
#43 0x00003dfb0c4a0ed9 in ?? ()
#44 0x0000394fc09b8a29 in ?? ()
#45 0x00007fffffffa1c8 in ?? ()
#46 0x0000276c6f353c94 in ?? ()
#47 0x000000e0b91ad019 in ?? ()
#48 0x00001c6f3800ea51 in ?? ()
#49 0x00003dfb0c4a0ed9 in ?? ()
#50 0x00003dfb0c4a17d1 in ?? ()
#51 0x000000e0b91afc99 in ?? ()
#52 0x00007fffffffa200 in ?? ()
#53 0x0000276c6f009895 in ?? ()
#54 0x000000e0b91ad019 in ?? ()
#55 0x00001c6f3800ea51 in ?? ()
#56 0x0000000200000000 in ?? ()
#57 0x00003dfb0c4a17d1 in ?? ()
#58 0x0000000e00000000 in ?? ()
#59 0x00007fffffffa240 in ?? ()
#60 0x0000276c6f35383d in ?? ()
#61 0x000011a780504201 in ?? ()
#62 0x000000e0b91ad019 in ?? ()
#63 0x00001c6f3800ea51 in ?? ()
#64 0x00003dfb0c4a17d1 in ?? ()
#65 0x00003dfb0c4a5601 in ?? ()
#66 0x0000394fc09bc441 in ?? ()
#67 0x00007fffffffa280 in ?? ()
#68 0x0000276c6f353699 in ?? ()
#69 0x000000e0b91ad019 in ?? ()
#70 0x000000e0b91acfe9 in ?? ()
#71 0x00001c6f38018af1 in ?? ()
#72 0x00003dfb0c4a5601 in ?? ()
#73 0x00001c6f380416a9 in ?? ()
#74 0x00001c6f3803f249 in ?? ()
#75 0x00007fffffffa2e8 in ?? ()
#76 0x0000276c6f28189a in ?? ()
#77 0x000000e0b91ad019 in ?? ()
#78 0x000000e0b91acfe9 in ?? ()
#79 0x00001c6f3803d299 in ?? ()
#80 0x000011a7805bd809 in ?? ()
#81 0x0000000100000000 in ?? ()
#82 0x0000000000000000 in ?? ()
```

**Comment by melv-n** [comment]
I experienced something similar in https://github.com/js-platform/node-webrtc/issues/278

**Comment by silverbucket** [comment]
It should be noted that this appears to just be happening on the develop branch and does not occur in the v0.0.61 release.

**Comment by markandrus** [comment]
I think this is fixed in v0.4.0-beta1.
